ntbs-service/Services/AlertService.cs

```diff
@@ -118,24 +115,42 @@ public async Task DismissAllOpenAlertsForNotification(int notificationId)
 
             foreach (var alert in alerts)
             {
+                // notification may have a duplicate alert to be dismissed on multiple other records
                 if (alert is DataQualityPotentialDuplicateAlert duplicateAlert)
                 {
-                    var duplicate =
-                        await _notificationRepository.GetNotificationForAlertCreationAsync(duplicateAlert.DuplicateId);
-
-                    if (duplicate.NotificationStatus.IsOpen())
-                    {
-                        continue;
-                    }
+                    await DismissDuplicationAlertsOnDuplicateRecord(notificationId, duplicateAlert.DuplicateId);
                 }
-
-                alert.ClosureDate = DateTime.Now;
-                alert.AlertStatus = AlertStatus.Closed;
+                
+                SetClosedAndAddClosureDate(alert);
             }
 
             await _alertRepository.SaveAlertChangesAsync();
         }
 
+        private async Task DismissDuplicationAlertsOnDuplicateRecord(int notificationId, int duplicateAlertId)
+        {
+            var duplicateRecordAlerts = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
+            if (!(duplicateRecordAlerts is null))
```

Comment:
I don't think duplicateRecordAlerts should ever be null itself. It is a list of alerts, so should either be empty or have at least 1 alert in.

Reply:
I think I've ended up with a neater solution using the null conditional operator which avoids having to do an explicit .Any() or == null check